tests/functional/test_fit_per_framework.py

```diff
@@ -712,6 +716,8 @@ def test_fit_sh(
                 env["CLASS_LABELS_FILE"] = f.name
 
         if parameters:
+            # I don't understand why hypersearch params are taken from a separate json
```

Comment:
This looks like a good discussion to have.  let's do it in conversation here, but please remove this comment when we are done.

Reply:
Let's do this...
So I still don't understand why json is needed there :)

Reply:
this looks like it's setting the actual values to run hyper parameters on.  model metadata contains the definition for the hyper parameters with  allowed values, etc.  but this actually sets the hyperparameters search space.

Reply:
it's very much like launch configuration.  the model metadata tells you what launch configs are allowed and what the defaults are, and then, when running drum, you can pass in actual set values via a json file.

Reply:
updated comment